crates/oxc_linter/src/rules/typescript/prefer_nullish_coalescing.rs

```diff
@@ -0,0 +1,160 @@
+use oxc_macros::declare_oxc_lint;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    rule::{DefaultRuleConfig, Rule},
+    utils::default_true,
+};
+
+#[derive(Debug, Default, Clone)]
+pub struct PreferNullishCoalescing(Box<PreferNullishCoalescingConfig>);
+
+/// Options for ignoring specific primitive types.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, JsonSchema, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct IgnorePrimitivesOptions {
+    /// Ignore bigint primitive types.
+    #[serde(default)]
+    pub bigint: bool,
+    /// Ignore boolean primitive types.
+    #[serde(default)]
+    pub boolean: bool,
+    /// Ignore number primitive types.
+    #[serde(default)]
+    pub number: bool,
+    /// Ignore string primitive types.
+    #[serde(default)]
+    pub string: bool,
+}
```

Comment:
Is there a reason not to do this as:

```suggestion
#[serde(rename_all = "camelCase", default)]
pub struct IgnorePrimitivesOptions {
    /// Ignore bigint primitive types.
    pub bigint: bool,
    /// Ignore boolean primitive types.
    pub boolean: bool,
    /// Ignore number primitive types.
    pub number: bool,
    /// Ignore string primitive types.
    pub string: bool,
}
```